### PR TITLE
Feature/auth behavior script

### PR DESCRIPTION
**Pull request by yuqche**
Add UI script for Auth and setConfigUrl for running and running and configuring Auth

Modify Android Content provider to include Auth

[AB#66424](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/66424)
[AB#66430](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/66430)

### CODE REVIEW

File: Assets/AppCenter/AppCenterSettings.cs [hunk 1]
```diff
@@ -23,12 +23,17 @@ public class AppCenterSettings : ScriptableObject
...
+    [Tooltip("App Center Auth is a cloud-based identity management service that enables developers to authenticate application users and manage user identities.")]
+    public bool UseAuth = true;
+
     [Tooltip("App Center Crashes will automatically generate a crash log every time your app crashes.")]
     public bool UseCrashes = true;
 
     [Tooltip("App Center Distribute will let your users install a new version of the app when you distribute it via the App Center.")]
     public bool UseDistribute = true;
 
+    public CustomUrlProperty CustomConfigUrl = new CustomUrlProperty("Config");
```
Comment (MatkovIvan) [inline Assets/AppCenter/AppCenterSettings.cs]: Let's add some prefix to this one

File: Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs [hunk 2]
```diff
@@ -17,6 +17,9 @@ public class AppCenterSettingsMakerAndroid : IAppCenterSettingsMaker
     private const string UseCrashesKey = "appcenter_use_crashes";
     private const string UsePushKey = "appcenter_use_push";
     private const string UseAnalyticsKey = "appcenter_use_analytics";
+    private const string UseAuthKey = "appcenter_use_auth";
+    private const string CustomConfigUrlKey = "appcenter_custom_config_url";
+    private const string UseCustomConfigUrlKey = "appcenter_use_custom_config_url";
```
Comment (MatkovIvan) [inline Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs]: It should be `appcenter_auth_*`

File: Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs [hunk 3]
```diff
@@ -82,6 +85,17 @@ public void StartAnalyticsClass()
         _resourceValues[UseAnalyticsKey] = true.ToString();
     }
 
+    public void StartAuthClass()
+    {
+        _resourceValues[UseAuthKey] = true.ToString();
+    }
+
+    public void SetConfigUrl(string configUrl)
```
Comment (MatkovIvan) [inline Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs]:
```suggestion
    public void SetAuthConfigUrl(string configUrl)
```

File: Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs (hunk 2 above)
Comment (yuqche) [reply Assets/AppCenter/Editor/AppCenterSettingsMakerAndroid.cs]: Sure I'll add the auth prefix to all uses of the config url

File: Assets/AppCenter/link.xml [hunk 4]
```diff
@@ -8,6 +8,7 @@
     <!-- Keep all our assemblies. -->
     <assembly fullname="Microsoft.AppCenter" preserve="all" ignoreIfMissing="1"/>
     <assembly fullname="Microsoft.AppCenter.Analytics" preserve="all" ignoreIfMissing="1"/>
+    <assembly fullname="Microsoft.AppCenter.Auth" preserve="all" ignoreIfMissing="1"/>
```
Comment (jaeklim) [inline Assets/AppCenter/link.xml:11]: @annakocheshkova Do you know if we need this one here?

Comment (annakocheshkova) [reply Assets/AppCenter/link.xml:11]: No it's for UWP afair

Comment (jaeklim) [reply Assets/AppCenter/link.xml:11]: This will be fixed in the next PR.

**Review by jaeklim — APPROVED** [review]

**Pull request merged**